src/modules/hip/kernel/mel_filter_bank.hpp

```diff
@@ -0,0 +1,155 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ __forceinline__ void compute_mel(float *srcPtr, int melBin, float *weightsDown, int *intervals, int2 fftStrides, float normFactor, float &dstVal)
+{
+    dstVal = 0;
+    //start and end FFT bin indices for the current mel bin
+    int fftbin = intervals[melBin];
+    int fftBinEnd = intervals[melBin + 1];
+
+    float *srcPtrTemp = srcPtr + fftbin * fftStrides.x + fftStrides.y;
+    // Process the first interval of FFT bins, applying the weights up
+    for (; fftbin < fftBinEnd; fftbin++, srcPtrTemp += fftStrides.x) 
+    {
+        auto weightUp = float(1) - weightsDown[fftbin];
```

Comment:
can we use float datatype, instead of auto datatype here?

Reply:
Done